v2/pkg/parsers/parser.go

```diff
@@ -68,10 +80,11 @@ func isTemplateInfoMetadataMatch(tagFilter *filter.TagFilter, template *template
 	return match, err
 }
 
-func validateTemplateFields(template *templates.Template) error {
+func validateTemplateFields(template *templates.Template) (err error, warning error) {
```

Comment:
What about splitting the function in `validateTemplateMandatoryFields` and `validateTemplateOptionalFields`?

Reply:
sounds nice... will update the functionality to split it